src/api/procedures/setTransferRestrictions.ts

```diff
@@ -64,6 +64,76 @@ export interface Storage {
   currentStats: BTreeSetStatType;
 }
 
+/**
+ * @hidden
+ */
+function isSameCondition(
+  transferCondition: TransferCondition,
+  inputCondition: CountTransferRestrictionInput | PercentageTransferRestrictionInput,
+  type: TransferRestrictionType
+): boolean {
+  if (transferCondition.isMaxInvestorCount && type === TransferRestrictionType.Count) {
+    const currentCount = u64ToBigNumber(transferCondition.asMaxInvestorCount);
+    return currentCount.eq((inputCondition as CountTransferRestrictionInput).count);
+  } else if (
+    transferCondition.isMaxInvestorOwnership &&
+    type === TransferRestrictionType.Percentage
+  ) {
+    const currentOwnership = permillToBigNumber(transferCondition.asMaxInvestorOwnership);
+    return currentOwnership.eq((inputCondition as PercentageTransferRestrictionInput).percentage);
+  } else {
+    return false;
+  }
+}
+
+/**
+ * @hidden
+ */
+function transformRestrictions(
+  restrictions: CountTransferRestrictionInput[] | PercentageTransferRestrictionInput[],
+  currentRestrictions: TransferCondition[],
+  type: TransferRestrictionType,
+  context: Context
+): [PolymeshPrimitivesTransferComplianceTransferCondition[], (string | Identity)[]] {
+  const exemptions: (string | Identity)[] = [];
+
+  let someDifference = restrictions.length !== currentRestrictions.length;
+  const conditions = restrictions.map(r => {
+    const compareConditions = (transferCondition: TransferCondition) =>
+      isSameCondition(transferCondition, r, type);
+    if (!someDifference) {
+      someDifference = !currentRestrictions.find(compareConditions);
+    }
+    let condition: TransferRestriction;
+    if (type === TransferRestrictionType.Count) {
+      condition = { type, value: (r as CountTransferRestrictionInput).count };
+    } else {
+      condition = { type, value: (r as PercentageTransferRestrictionInput).percentage };
+    }
```

Comment:
I see what you mean, but the `SetCountTransferRestrictionsParams` has restrictions as an Array, where as I have only a single value I am feeding in, so it doesn't quite work.

I was able to change this a bit to eliminate one of the assertions.

What I really want is for TS to let me do something like

`const value = inputCondition.count || inputCondition.percentage`, since I know one of them will be defined.

Reply:
What you can do is 

```
let value: BigNumber;

if ('count' in inputCondition) {
  value = inputCondition.count;
} else {
  value = inputCondition.percentage;
}
```

You could stick this in a ternary as well but I'd rather not